### PR TITLE
Feature/270 add enums for analysis config settings

### PR DESCRIPTION
**Pull request by ArdtK**
Made 2 enums: AnalysisEnumDirect and AnalysisEnumIndirect.
Did some refactoring so list_valid_options is a classmethod, which makes more sense to me.

### CODE REVIEW

File: ra2ce/analyses/analysis_config_data/analysis_config_data.py [hunk 1]
```diff
@@ -36,18 +42,11 @@
...
-    "optimal_route_origin_destination",
-    "multi_link_origin_destination",
-    "optimal_route_origin_closest_destination",
-    "multi_link_origin_closest_destination",
-    "losses",
-    "single_link_losses",
-    "multi_link_losses",
-    "multi_link_isolated_locations",
+    _enum.config_value for _enum in AnalysisIndirectEnum.list_valid_options()
+]
```
Comment (Carsopre) [inline ra2ce/analyses/analysis_config_data/analysis_config_data.py:46]: based on what you've implemented you can simply:
```python
list(map(str, AnalysisIndirectEnum.list_valid_options))
```
Of course, test this :)

File: ra2ce/analyses/analysis_config_data/analysis_config_data_validator.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def _validate_header(self, header: Any) -> ValidationReport:
                     continue
                 if isinstance(value, Ra2ceEnumBase):
                     # enumerations
-                    _expected_values_list = value.list_valid_options()
+                    _expected_values_list = type(value).list_valid_options()
```
Comment (Carsopre) [inline ra2ce/analyses/analysis_config_data/analysis_config_data_validator.py:59]: This should not be needed when working with a classmethod, the object itself will still have access to it.

File: ra2ce/graph/network_config_data/network_config_data_validator.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def _wrong_value(self, key: str) -> str:
 
     def _wrong_enum(self, key: str, enum: Ra2ceEnumBase) -> str:
         # Remove last value INVALID
-        _accepted_values = enum.list_valid_options()
+        _accepted_values = type(enum).list_valid_options()
```
Comment (Carsopre) [inline ra2ce/graph/network_config_data/network_config_data_validator.py:81]: `type` here should not be needed. If so, there's something wrong in the code.
https://docs.python.org/3/library/functions.html#classmethod

**Review by Carsopre — CHANGES_REQUESTED** [review]

File: ra2ce/analyses/analysis_config_data/analysis_config_data.py (hunk 1 above)
Comment (ArdtK) [reply ra2ce/analyses/analysis_config_data/analysis_config_data.py:46]: Nice suggestion!

**Review by Carsopre — APPROVED** [review]

**Pull request merged**